### ISSUE TITLE
[java] Cognitive Complexity threshold cannot be set to 0

### ISSUE THREAD

**Issue by edoardosarri24** [pmd/pmd#5917]
**Affects PMD Version:** 7.15.0

**Rule:** [CognitiveComplexity](https://docs.pmd-code.org/latest/pmd_rules_java_design.html#cognitivecomplexity)

**Running PMD through:** CLI

**Description:** The threshold value must be $s>0$. If $s=0$, the analysis does not return any files. For this reason, very simple Java files cannot be introduced into the final report.

**Code Sample:** The class below isn't include in the final PMD report.
```java
public class SampleDuration {
    public static Duration sample(Sampler sampler) {
        BigDecimal sample = sampler.getSample();
        sample = sample.multiply(BigDecimal.TEN.pow(6));
        return Duration.ofNanos(sample.longValue());
    }
}
```

**Comment by adangel** [comment]
Do I understand this correctly? Your use case is not finding complex classes, instead it is getting the metric for each class?

This reminds me of #5166 ...

**Comment by edoardosarri24** [comment]
Yes, exactly. This can be useful when you need to correlate metrics like CC with other information to obtain a complex report. In my case study, for example, I correlate each file with the contribution percentage of each developer that worked on that file and the CC of the file (e.g., like [this](https://github.com/user-attachments/files/21322316/tltc_results.csv) csv).

My idea is that if 1 is allowed as a threshold value, why not also allow 0? In situations where you only want to find complex classes, 1 is probably too low a threshold value and not very interesting. Introducing 0 as a threshold value makes it easier to merge multiple analyses (as in my case study).
